be/src/io/cache/block_file_cache.cpp

```diff
@@ -1807,6 +1811,61 @@ void BlockFileCache::update_ttl_atime(const UInt128Wrapper& hash) {
     };
 }
 
+std::string BlockFileCache::clear_file_cache_by_key(bool sync, const UInt128Wrapper& hash) {
+    std::lock_guard cache_lock(_mutex);
+    std::stringstream ss;
+    LOG(INFO) << "start clear_file_cache_by_key sync=" << sync;
+
+    if (!_files.contains(hash)) {
+        LOG(INFO) << "no file need to be deleted, hash=" << hash.to_string();
+    }
+
+    std::vector<Path> files;
+    for (auto& [offset, cell] : _files[hash]) {
+        std::string file =
+                cell.file_block->_mgr->_storage->get_local_file(cell.file_block->_key);
```

Comment:
We will have a couple of new Storage types that do not use the file system as the cache storage medium, e.g. In-memory storage that only uses system RAM. So make this interface more general, not only for 'local file'. IMHO, there should be a storage->delete(key) interface instead to keep FSFileCacheStorage's 'file' unexposed to upper class BlockFileCache.